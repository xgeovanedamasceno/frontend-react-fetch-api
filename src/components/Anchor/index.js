import PropTypes from 'prop-types';
import AnchorStyle from './styled';

function Anchor({ url, name }) {
  return (
    <AnchorStyle>
      <a href={url}>{ name }</a>
    </AnchorStyle>
  );
}

Anchor.propTypes = {
  url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

export default Anchor;
