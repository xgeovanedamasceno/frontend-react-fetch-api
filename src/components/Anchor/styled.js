import styled from 'styled-components';

const AnchorStyle = styled.span`
  background-color: #F6F8FA;
  border-radius: 5px;
  color: #24292F;
  padding: 16px 8px;
`;

export default AnchorStyle;
