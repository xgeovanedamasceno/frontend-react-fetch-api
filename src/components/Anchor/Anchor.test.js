import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Anchor from '.';

describe('Anchor', () => {
  beforeEach(() => {
    render(<Anchor url="#exemplo" name="Products" />);
  });

  it('should render an anchor element with the text "Products"', () => {
    const anchorEl = screen.getByRole('link', { name: 'Products' });

    expect(anchorEl).toBeInTheDocument();
  });

  it('should render an anchor element with href value "#exemplo"', () => {
    const anchorEl = screen.getByRole('link', { name: 'Products' });

    expect(anchorEl).toHaveAttribute('href', '#exemplo');
  });

  it('should render correctly', () => {
    const tree = renderer
      .create(<Anchor url="#exemplo" name="Products" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
