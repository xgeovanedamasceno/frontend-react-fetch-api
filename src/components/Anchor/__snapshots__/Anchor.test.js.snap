// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor should render correctly 1`] = `
.c0 {
  background-color: #F6F8FA;
  border-radius: 5px;
  color: #24292F;
  padding: 16px 8px;
}

<span
  className="c0"
>
  <a
    href="#exemplo"
  >
    Products
  </a>
</span>
`;
